### DISCUSSION TITLE
cc64 v0.9

### DISCUSSION THREAD

**Discussion by pzembrod** [pzembrod/cc64#26] [General]
Main topic of v0.9 is compile speed. With the help of a simple profiler
I was able to optimize a few performance hotspots in the scanner and in
the scanner/parser interface, resulting in more than 20% compile time
saved or more than 30% speed gained.

Also, Source code listing during compile can now be switched on and off.
Default is off; it turns out that switching off listing decreases
compile time by another ~10%.

<hr /><em>This discussion was created from the release <a href='https://github.com/pzembrod/cc64/releases/tag/v0.9'>cc64 v0.9</a>.</em>

**Comment by jjcale** [comment]
This is a very nice attempt to bring a C compiler to the Commodore Plus/4 world. I tested it with YAPE and it works!
Wish I had that in 1987 ;-).
BTW, where can I find documentation for the commands (that I can list with HELP)?
Are there any libraries developed for Plus/4?

update:
I found the link for the commands:
https://github.com/pzembrod/cc64/blob/master/Usage.md

**Reply by pzembrod** [reply]
Glad you like it, and glad you found the Usage page. Agree about 1987 ... :-)
And I'm not aware of any Plus/4-libraries in development at this time. What kind of libraries would you be interested in?

**Reply by jjcale** [reply]
I was wondering if there are any libraries that could be used at all. One would have to start with KERNAL i/o functions, then simple graphics i/o functions, etc. Possibly some could be ported from C64.

**Reply by pzembrod** [reply]
Getting back to this after a while:
For basic Kernal I/O the kernal-io-c16.c in src/samples may already do a bit of what you are looking for. Some graphics functions of course would also be nice.
I just started to port PDCLib (at https://pdclib.rootdirectory.de), and esp. its tests, to cc64. I'll see how far I can make it work. Hat tip at this point to Martin Baute, PDCLib's author and maintainer.
Another effort can be found at https://sourceforge.net/projects/cc64/ where buzzphp has written their own libc for cc64.

**Reply by pzembrod** [reply]
I just released v0.10, which has a basic LibC.
